unity-renderer/Assets/Scripts/MainScripts/DCL/Components/TextShape/TextShape.cs

```diff
@@ -62,8 +62,14 @@ public class Model : BaseModel
         public TextMeshPro text;
         public RectTransform rectTransform;
         private Model cachedModel;
+        private Material cachedFontMaterial;
 
-        private void Awake() { model = new Model(); }
+        private void Awake()
+        {
+            model = new Model();
+            cachedFontMaterial = new Material(text.fontSharedMaterial);
+            text.fontSharedMaterial = cachedFontMaterial;
```

Comment:
Probably is cleaner this way:
```suggestion
            cachedFontMaterial = text.fontMaterial;
```
since `text.fontMaterial` is already creating a new material and assigning it to the `TextMeshPro`

Reply:
I'd prefer not relying on the `fontMaterial` awful copying mechanism. It's easier to understand as it is now.